src/AppBundle/Resources/public/js/app.deck.js

```diff
@@ -1017,11 +1040,17 @@ deck.reset_limit_count = function (){
 				deck.deck_options[i].atleast_count = {};
 			}
 		}
+
 		if (on_your_own && on_your_own.indeck) {
 			// We put it at the front of the requirements since it is a negated one.
 			var new_option = {name: "on_your_own", dynamic: true, not: true, slot: ['Ally']};
 			deck.deck_options.unshift(new_option);
 		}
+		if (ancestral_knowledge && ancestral_knowledge.indeck) {

```

Comment:
One bit of subtleness, it has to go towards the front of the list so that it counts everything.

This does mean it might count a card that is later rejected, but if a card is 'rejected' for matching a NOT clause (or not matching any clause), the deck will fail validation for that reason.